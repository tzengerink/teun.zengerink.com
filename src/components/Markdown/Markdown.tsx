import React from 'react'
import { marked } from 'marked'
import classNames from 'classnames'

interface Props {
  children: React.ReactNode
}

const toInnerHtml = (
  markdown: string,
): {
  __html: string
} => ({ __html: marked.parse(markdown) })

const Markdown: React.FC<Props> = ({ children }) => {
  return (
    <div
      className={classNames('space-y-sm', 'text-xs', 'text-justify')}
      dangerouslySetInnerHTML={toInnerHtml(children.toString())}
    />
  )
}

export default Markdown
