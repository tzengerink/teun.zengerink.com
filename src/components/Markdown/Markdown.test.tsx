import { render } from '@testing-library/react'
import Markdown from './Markdown'

describe('Markdown', () => {
  it('renders correctly', () => {
    const markdown = `The text **that** contains _some_ markdown.
    
It also supports [multiple](http://teun.zengerink.com/) paragraphs.`
    const { container } = render(<Markdown>{markdown}</Markdown>)
    expect(container).toMatchSnapshot()
  })
})
