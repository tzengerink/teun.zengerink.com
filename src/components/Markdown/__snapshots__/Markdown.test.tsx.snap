// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown renders correctly 1`] = `
<div>
  <div
    class="space-y-sm text-xs text-justify"
  >
    <p>
      The text 
      <strong>
        that
      </strong>
       contains 
      <em>
        some
      </em>
       markdown.
    </p>
    

    <p>
      It also supports 
      <a
        href="http://teun.zengerink.com/"
      >
        multiple
      </a>
       paragraphs.
    </p>
    

  </div>
</div>
`;
